pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/participantdata/RgpParticipantDataService.java

```diff
@@ -240,4 +245,15 @@ private static String getPhoneFromActivities(List<Activities> activities) {
             return phoneAnswer.map(answer -> answer.get(DDPActivityConstants.ACTIVITY_QUESTION_ANSWER)).orElse("");
         }).orElse("");
     }
+
+    // TODO: remove this method after debugging is complete. -DC
+    private static void logParticipantDocument(String ddpParticipantId, String esIndex) {
+        Optional<String> ptpDoc =
+                ElasticSearchService.getParticipantDocumentAsString(ddpParticipantId, esIndex);
```

Comment:
Any concerns about the size of this string, or whether the inability to hit ES to get the data may itself cause an exception that prevents existing log messages (like in L164)?

Reply:
Not really. It's possible to IOExceptions from ES, but that would likely be true of the ES call that DSM just made to get the profile data. The case of a missing document is handled.

The string will easily fit in memory.
I've used a version of this is testing for a while without problems.

Worse case is this code throws an exception, which is handled but ends the retry sequence. Our ace is that we now have an AdminOperation to fix cases that go wrong, which we will use to resolve the existing Prod error cases.

Does that make sense?